crates/iceberg/src/arrow/reader.rs

```diff
@@ -1180,4 +1194,62 @@ mod tests {
 
         assert_eq!(visitor.field_ids, expected);
     }
+
+    #[test]
+    fn test_arrow_projection_mask() {
```

Comment:
Hi, thank you @gruuya for working on this, would you like to add a test case for decimal? Others LGTM.

Reply:
Certainly, let me think of one.

Reply:
I've slightly revised the existing test now. It tests a parquet/arrow schema with one supported type, one unsupported type and one with a decimal with a precision beyond the supported range. It asserts that projection masks fails until the projection leaves out the last two fields.

Granted it's a bit of a stretch to think how this could come up in practice, but it does exercise the proposed changes in this PR nonetheless.